### DISCUSSION TITLE
Refund makes an order go back into processing?

### DISCUSSION THREAD

**Discussion by fballiano** [OpenMage/magento-lts#1863] [Questions & Answers]
Hi everybody,
I've been reported something weird happening on a customer's project, when refunding the order goes back to processing state. I've been checking the thing and I've found that https://github.com/OpenMage/magento-lts/blob/Flyingmana-patch-1/app/code/core/Mage/Sales/Model/Order/Payment.php (line 725) is actally doing exactly that, during the refund method it resets the order state to processing.

Does it make sense to you? Am I missing something? Cause to me that's not the right behaviour.
Thanks everybody!

**Comment by justinbeaty** [comment]
That would seem to not be the right behavior. I asked one of my store owners and she said she's never seen that happen though. She always checked to make sure it says closed. But she will keep an eye out to make sure the order wouldn't get shipped again.

Edit: git blame says that's from version 1.4.0. Maybe there is something else that sets it to complete, like a sales_order_payment_refund observer?

**Comment by elidrissidev** [comment]
Yes it does go into `processing` state, but it should go to `closed` state directly after as you can see from the snippet below the method `_checkState`which  is called before saving the order, so maybe one of the conditions fails (Line 2383):
https://github.com/OpenMage/magento-lts/blob/8b9bfae16c35f2740b7e066864ee07a5db64218a/app/code/core/Mage/Sales/Model/Order.php#L2359-L2392

Here is a test I did in a clean installation:
![Order refund test](https://user-images.githubusercontent.com/67818913/138521047-e460f772-63a2-4a13-a608-73cc79616e9f.png)

Let me know if it helps.
